src/Components/VoteOnArticle.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import { patchVotesOnArticle } from "../api";
+
+const VoteOnArticle = ({ votes }) => {
+  const [articleVotes, setArticleVotes] = useState(0);
+  const [buttonAddVotesDisabled, setButtonAddVotesDisabled] = useState(false);
+  const [buttonDeleteVotesDisabled, setButtonDeleteVotesDisabled] =
+    useState(true);
+  const [error, setError] = useState(null);
+  const { article_id } = useParams();
+
+  let totalVotes = votes + articleVotes;
+
+  const handleAddVotesonArticle = (change) => {
+    setButtonAddVotesDisabled(true);
+    setButtonDeleteVotesDisabled(false);
+    setArticleVotes((currVotes) => currVotes + change);
+    patchVotesOnArticle(article_id, { inc_votes: change }).catch(() => {
+      setError("This page is not working now.Try later!");
+    });
+  };
+
+  const handleDeleteVotesonArticle = (change) => {
+    setButtonDeleteVotesDisabled(true);
+    setArticleVotes((currVotes) => currVotes - change);
+    patchVotesOnArticle(article_id, { inc_votes: change }).catch(() => {
+      setError("This page is not working now.Try later!");
+    });
+  };
```

Comment:
This should be refactored to be one function only to be more DRY

Reply:
It was just a function but as I was suggested to be able to disable both buttons, I did not found another way to do so. From the point of view of functionality, I will leave it like that. I also want to prevent trolling so you can only vote or delete once. But I understand your comments. If it is alright. I will close this ticket